pkg/kubelet/kubelet_pods.go

```diff
@@ -1756,7 +1757,10 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	}
 	s := kl.convertStatusToAPIStatus(pod, podStatus, oldPodStatus)
 	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
-		s.Resize = kl.determinePodResizeStatus(pod, podStatus, podIsTerminal)
+		resizeStatus := kl.determinePodResizeStatus(pod, podStatus, podIsTerminal)
+		for _, condition := range resizeStatus {
+			podutil.UpdatePodCondition(s, condition)
```

Comment:
This should be done after `s.Conditions` is initialized:
https://github.com/kubernetes/kubernetes/blob/2e9bb32ee8defbd37ce9754ae8362356ce4c3796/pkg/kubelet/kubelet_pods.go#L1824